flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.highavailability;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.HighAvailabilityOptions;
+import org.apache.flink.runtime.clusterframework.ApplicationStatus;
+import org.apache.flink.runtime.jobmaster.JobResult;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+/** Tests for the {@link FileSystemJobResultStore}. */
+public class FileSystemJobResultStoreTest extends TestLogger {
```

Comment:
I remember parts of the code. :-D Initially, I was wondering whether we should combine `EmbeddedJobResultStoreTest` and `FileSystemJobResultStoreTest` to reduce code redundancy. But I'm not a big fan of coming up with abstract test classes because it makes reading the test class harder. Alternatively, we could refactor `EmbeddedJobResultStore` to provide static test method that can be called within `EmbeddedJobResultStoreTest` and `FileSystemJobResultStoreTest`. Not sure, maybe, @dmvk has a strong opinion on that one.

Anyway, the `FileSystemJobResultStoreTest` needs to check the actual file creation as well since it's part of the classes state.

Reply:
Yeah, I think coming up with an abstract test class for only 2 test classes is maybe overkill. Could be something to look forward to if we implement another implementation down the line.